src/gui/lineedit/src/lineedit.h

```diff
@@ -12,26 +12,29 @@
 
 #include <QLineEdit>
 
-QT_BEGIN_NAMESPACE
 class QToolButton;
-QT_END_NAMESPACE
 
 class LineEdit : public QLineEdit
 {
   Q_OBJECT
 
 public:
-  LineEdit(QWidget *parent = 0);
```

Comment:
>it's bad practice to default to 0

It is a good practice if it makes sense for the class itself. QWidget without parent it's correct use case. If you want some particular object had a parent, then you just need to call the constructor with the appropriate parameter in this place.
OOP is not intended to protect the programmer from wrong actions (it's impossible).